eppo_core/src/background/runtime.rs

```diff
@@ -34,73 +35,70 @@ pub struct BackgroundRuntime {
         let watched_tasks = TaskTracker::new();
         BackgroundRuntime {
-            tokio_runtime: runtime,
+            async_runtime: runtime,
             cancellation_token,
             watched_tasks,
         }
     }
 
-    #[must_use]
-    pub(crate) fn tokio_handle(&self) -> &tokio::runtime::Handle {
-        &self.tokio_runtime
-    }
-
-    /// Returns a cancellation token that get cancelled when background runtime is going to
-    /// shutdown.
-    ///
-    /// If you want your task to react to shutdown and perform some cleanup, spawn your task with
-    /// [`BackgroundThread::spawn_tracked()`] and watch this cancellation token.
-    ///
-    /// Note: this is a child token of the main cancellation token, so canceling it does not cancel
-    /// the whole runtime.
-    #[must_use]
-    pub(crate) fn cancellation_token(&self) -> CancellationToken {
-        self.cancellation_token.child_token()
-    }
-
-    pub fn block_on<F: Future>(&self, future: F) -> F::Output {
-        self.tokio_runtime.block_on(future)
-    }
-
     /// Spawn a task that needs to perform some cleanup on shutdown.
     ///
     /// Most tasks shouldn't need that as Rust futures are usually safe to drop.
     ///
     /// The task must monitor [`BackgroundRuntime::cancellation_token()`] and exit when the token is
     /// cancelled.
-    pub(crate) fn spawn_tracked<F>(&self, future: F) -> JoinHandle<F::Output>
+    pub(crate) fn spawn_tracked<F>(&self, future: F)
     where
         F: Future + Send + 'static,
         F::Output: Send + 'static,
     {
-        self.tokio_runtime
-            .spawn(self.watched_tasks.track_future(future))
+        self.async_runtime
+            .spawn(self.watched_tasks.track_future(future));
     }
 
     /// Spawn a task that doesn't have any special shutdown requirements.
     ///
     /// When runtime is going to shutdown, this task will not be awaited and will be abandoned.
     ///
     /// If it's not OK to abandon the task, consider using `spawn_tracked()` instead.
-    pub(crate) fn spawn_untracked<F>(&self, future: F) -> JoinHandle<F::Output>
+    pub(crate) fn spawn_untracked<F>(&self, future: F)
     where
         F: Future + Send + 'static,
         F::Output: Send + 'static,
     {
-        self.tokio_runtime.spawn(future)
+        let cancellation_token = self.cancellation_token().clone();
+        self.async_runtime.spawn(async move {
+            cancellation_token.run_until_cancelled(future).await;
+        });
+    }
+}
+
+impl<AR> BackgroundRuntime<AR> {
+    /// Returns a cancellation token that get cancelled when background runtime is going to
+    /// shutdown.
+    ///
+    /// If you want your task to react to shutdown and perform some cleanup, spawn your task with
+    /// [`BackgroundThread::spawn_tracked()`] and watch this cancellation token.
+    ///
+    /// Note: this is a child token of the main cancellation token, so canceling it does not cancel
```

Comment:
very nice - this will be primarily used within SDKs or can you imagine a customer facing use too?

Reply:
I don't think there's customer use. Right now, this is only used within core (not even exported to individual sdks)